### DISCUSSION TITLE
Maybe a request for SVG files?

### DISCUSSION THREAD

**Discussion by NewtonChutney** [lunapaint/vscode-luna-paint#207] [General]
@Tyriar, I've been using your extension for a while now.. just love it! ❤️‍🔥

I was recently building a binary and trying to integrate an ico file into it.. But noticed Luna paint didn't preview SVG files..
Is it something about vector graphics, that you opted out of this category?

And if there isn't any blocker, I'd like this to be a feature request for conversions between multiple formats and support for SVG/vector graphics! 😶

**Comment by Tyriar** [comment]
Yes I excluded vector graphics completely as it's a very different beast. All of a sudden you're not dealing with pixels but draw operations, and while Luna Paint does handle some of that, it would complicate things quite a bit to try support both. I may at some point give vector graphics a shot as I've always been interested in that too, but not sure integrating it into Luna Paint would be the best idea 🤔

**Reply by NewtonChutney** [reply]
Yeah, vector graphics seem to be a different ecosystem too altogether..

**Reply by NewtonChutney** [reply]
And btw, what about conversions maybe between the other common formats? png to jpg, heic, ico.. Etc?

**Reply by Tyriar** [reply]
You can convert between the formats that the extension supports via Save As (eg. open a jpg, save as and add .webp).

I hadn't heard of heic before but that makes sense since I haven't owned an iphone/ipad before. I've been taking a break from Luna Paint for a while, but adding new formats was normally a project I'd do over several weekends building a codec like https://github.com/lunapaint/tga-codec

**Reply by NewtonChutney** [reply]
I'm sorry for assuming Luna paint didn't support exporting.. But when I click the menu button in the image editor, I get the command pallete filtered with the Luna keyword.. I don't see a save as option there.. 😶

And I tried the VSCode > File menu > Save as, but I see the file internally doesn't change formats..

**Reply by Tyriar** [reply]
Hmm, it worked for me just now when I changed the file extension:

![Recording 2023-12-18 at 10 11 14](https://github.com/lunapaint/vscode-luna-paint/assets/2193314/698ea209-a935-4779-b583-ff680f79b89e)

That is with `"files.simpleDialog.enable": true` set. You probably don't have that set, if you don't you would need to do the regular dance to save as a different file type based on your operating system. For example on Windows:

![Recording 2023-12-18 at 10 13 10](https://github.com/lunapaint/vscode-luna-paint/assets/2193314/f4690c66-81c6-47e3-9864-fe2c9c070e8b)

**Reply by NewtonChutney** [reply]
Yes, I tried saving into jpg from png through the save as dialog (selected all files type).. But when I used the `file` command from mingw.. I find the internal data isn't converted
